observer/probers/dns/dns_conf.go

```diff
@@ -126,6 +145,14 @@ func (c DNSConf) MakeProber() (probers.Prober, error) {
 	}, nil
 }
 
+// Instrument constructs any `prometheus.Collector` objects the `DNSProbe` will
+// need to report its own metrics. A map is returned containing the constructed
+// objects, indexed by the name of the prometheus metric. If no objects were
+// constructed, an empty map is returned.
+func (c DNSConf) Instrument() map[string]*prometheus.Collector {
+	return map[string]*prometheus.Collector{}
```

Comment:
Maps are an inherently nillable type, so it's more idiomatic to simply return `nil` here. If that means you have to do a `!= nil` check somewhere else, just do it :)

Reply:
Done!